src/internal/pfsdb/branches.go

```diff
@@ -146,6 +146,14 @@ func NewBranchIterator(ctx context.Context, tx *pachsql.Tx, startPage, pageSize
 	}, nil
 }
 
+func ListBranches(ctx context.Context, tx *pachsql.Tx, filter *pfs.Branch, orderBys ...OrderByBranchColumn) (*BranchIterator, error) {
```

Comment:
Hmm I was under the impression that `ListBranches` would do a bit more than just wrap the constructor. I think `ListBranches` should take in the user defined callback, and wrap the stream.ForEach instead.

```Go
func ListBranches(ctx context.Context, tx *pachsql.Tx, handleBranch func(branch BranchInfoWithID), filter *pfs.Branch, orderBys ...OrderByBranchColumn) error {
    iter, err := NewBranchIterator(ctx, tx, 0, 100, filter, orderBys...)
    stream.ForEach[BranchInfoWithID](ctx, handleBranch)
}
```

Reply:
feedback should be implemented as we discussed 👍